### PR TITLE
Use updated Vision libraries.

### PR DESCRIPTION
**Pull request by yunikkk**

### CODE REVIEW

File: app/src/main/java/com/mapbox/vision/examples/activity/main/MainActivity.kt [hunk 1]
```diff
@@ -92,11 +90,17 @@ class MainActivity : AppCompatActivity() {
 
     private val visionEventsListener = object : VisionEventsListener {
 
+        override fun onAuthorizationStatusUpdated(authorizationStatus: AuthorizationStatus) {
+            println("Auth $authorizationStatus")
```
Comment (RingerJK) [inline app/src/main/java/com/mapbox/vision/examples/activity/main/MainActivity.kt]: you'd like to have system out print here. Are you sure?

File: build.gradle [hunk 2]
```diff
@@ -11,9 +11,10 @@ buildscript {
     }
 
     dependencies {
-        classpath "com.android.tools.build:gradle:$gradle_plugin"
+        classpath "com.android.tools.build:gradle:3.4.1"
```
Comment (RingerJK) [inline build.gradle]: please change in _gradle_plugin_  property in `gradle.version`

File: gradle/versions.gradle [hunk 3]
```diff
@@ -13,17 +13,20 @@ ext {
     android_x_constraint_layout = '1.1.3'
     android_x_multidex = '2.0.1'
 
-    vision_native = '0.3.0'
-    vision_sdk = '0.3.0'
-    vision_ar = '0.3.0'
-    vision_safety = '0.3.0'
+    vision = '0.4.0'
+    spek_version = '1.0.0'
```
Comment (RingerJK) [inline gradle/versions.gradle]: spek version is 2.0.4 and please move it to the bottom of file

File: app/src/main/java/com/mapbox/vision/examples/activity/main/MainActivity.kt (hunk 1 above)
Comment (yunikkk) [reply app/src/main/java/com/mapbox/vision/examples/activity/main/MainActivity.kt]: Nope, sure, will remove, debug trash(

File: gradle/versions.gradle (hunk 3 above)
Comment (yunikkk) [reply gradle/versions.gradle]: Riiight, again - debug, wasn't really intended to push this yet.

**Comment by yunikkk** [comment]
@RingerJK ptal

**Pull request merged**